### PR TITLE
doc/dev,hack/tests: update docs and fix go test configuration

### PR DESCRIPTION
**Pull request by AlexNPavel**
**Description of the change:** Update testing documentation and fix ARGS handling for `make test/e2e/go`

**Motivation for the change:** Keep documentation up-to-date and make sure that test configuration works properly

### CODE REVIEW

**Comment by AlexNPavel** [comment]
@LiliC I noticed that the `travis-build.md` doc that describes what the tests do is missing the metrics tests. Could you make a PR to add those as well (or just push directly to my branch to update it with this PR, which you should have permission to since you're a project maintainer)

**Comment by LiliC** [comment]
@AlexNPavel Good catch! Done.

File: doc/dev/testing/travis-build.md [hunk 1]
```diff
@@ -44,24 +44,32 @@ The Go, Ansible, and Helm tests then differ in what tests they run.
...
+            2. Run the leader election test.
+                1. Verify that operator deployment is ready.
+                2. Verify that leader configmap specifies 1 leader and that the memcached operator has 2 pods (configuration for this is done in step 4.1).
+                3. Delete current leader and wait for memcached-operator deployment to become ready again.
+                4. Verify that leader configmap specifies 1 leader and that the memcached-operator has 2 pods.
+                5. Verify that the name of the new leader is different from the name of the old leader.
+            3. Run the memcached scale test.
+                1. Create memcached CR specifying a desired cluster size of 3 and wait until memcached cluster is of size 3.
+                2. Increase desired cluster size to 4 and wait until memcached cluster is of size 4.
+            3. Run the memcached metrics test.
```
Comment (hasbro17) [inline doc/dev/testing/travis-build.md]:
```suggestion
            4. Run the memcached metrics test.
```

File: doc/dev/testing/travis-build.md [hunk 2]
```diff
@@ -44,24 +44,32 @@ The Go, Ansible, and Helm tests then differ in what tests they run.
...
+                1. Make sure the metrics Service was created.
+                2. Get metrics via proxy pod and make sure they are present.
+                3. Perform linting of the existing metrics.
+            3. Run the memcached custom resource metrics test.
```
Comment (hasbro17) [inline doc/dev/testing/travis-build.md]:
```suggestion
            5. Run the memcached custom resource metrics test.
```

**Pull request merged**